src/plugins/dist_matrixops/dist_identity.cpp

```diff
@@ -0,0 +1,310 @@
+            for (std::size_t i = 0; i != column_size; ++i)
+            {
+                std::size_t j = i + column_start;
+                m(j, i) = 1.0;
+            }
+        }
+        else if (tiling_type == "sym" && numtiles == 4)
+        {
+            for (std::size_t i = 0; i != column_size; ++i)
+            {
+                std::int64_t j = i + column_start - row_start;
+                if (j >= row_size || j < 0)
+                {
+                    break;
+                }
+                m(j, i) = 1.0;
+            }
+        }
+        else
+        {
+            HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                "dist_identity::dist_identity_helper",
+                generate_error_message(
+                    "wrong numtiles input when tiling_type is sym"));
+        }
+
+        return primitive_argument_type(std::move(m), attached_annotation);
+    }
+
+    execution_tree::primitive_argument_type dist_identity::dist_identity_nd(
+        std::int64_t&& sz, std::uint32_t const& tile_idx,
+        std::uint32_t const& numtiles, std::string&& given_name,
+        std::string const& tiling_type,
+        execution_tree::node_data_type dtype) const
+    {
+        using namespace execution_tree;
+
+        switch (dtype)
+        {
+        case node_data_type_bool:
+            return dist_identity_helper<std::uint8_t>(std::move(sz), tile_idx,
+                numtiles, std::move(given_name), tiling_type);
+
+        case node_data_type_int64:
+            return dist_identity_helper<std::int64_t>(std::move(sz), tile_idx,
+                numtiles, std::move(given_name), tiling_type);
+
+        case node_data_type_unknown:
+            HPX_FALLTHROUGH;
+        case node_data_type_double:
+            return dist_identity_helper<double>(std::move(sz), tile_idx,
+                numtiles, std::move(given_name), tiling_type);
+
+        default:
+            break;
+        }
+
+        HPX_THROW_EXCEPTION(hpx::bad_parameter,
+            "dist_matrixops::dist_identity::dist_identity_nd",
+            util::generate_error_message(
+                "the constant primitive requires for all arguments to "
+                "be numeric data types"));
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    hpx::future<execution_tree::primitive_argument_type> dist_identity::eval(
+        execution_tree::primitive_arguments_type const& operands,
+        execution_tree::primitive_arguments_type const& args,
+        execution_tree::eval_context ctx) const
+    {
+        // verify arguments
+        if (operands.size() < 1 || operands.size() > 6)
+        {
+            HPX_THROW_EXCEPTION(hpx::bad_parameter, "dist_identity::eval",
+                generate_error_message("the identity_d primitive requires "
+                                       "at least 1 and at most 6 operands"));
+        }
+
+        if (!valid(operands[1]) || !valid(operands[2]) || !valid(operands[3]))
```

Comment:
I think as only the first operand should be given, we must check only `if(!valid(operands[0]))` (if I am not mistaken, the other are initialized and will be checked as args to see if a value is given or `nil` is assigned to them). 
I am sure this is due to my mistake in `random_d` (it should have been `operands[0]` instead of `operands[3]`) and sorry I wasn't clear about this.

Reply:
Changed to if(!valid(operands[0])).